agent/config/runtime_test.go

```diff
@@ -0,0 +1,3562 @@
+						"script": "MDu7wjlD",
+						"http": "qzHYvmJO",
+						"header": {
+							"UkpmZ3a3": ["2dfzXuxZ"],
+							"cVFpko4u": ["gGqdEB6k", "9LsRo22u"]
+						},
+						"method": "X5DrovFc",
+						"tcp": "ICbxkpSF",
+						"interval": "24392s",
+						"docker_container_id": "ZKXr68Yb",
+						"shell": "CEfzx0Fo",
+						"tls_skip_verify": true,
+						"timeout": "38333s",
+						"ttl": "57201s",
+						"deregister_critical_service_after": "44214s"
+					}
+				},
+				{
+					"id": "MRHVMZuD",
+					"name": "6L6BVfgH",
+					"tags": ["7Ale4y6o", "PMBW08hy"],
+					"address": "R6H6g8h0",
+					"token": "ZgY8gjMI",
+					"port": 38292,
+					"enable_tag_override": true,
+					"checks": [
+						{
+							"id": "GTti9hCo",
+							"name": "9OOS93ne",
+							"notes": "CQy86DH0",
+							"status": "P0SWDvrk",
+							"script": "6BhLJ7R9",
+							"http": "u97ByEiW",
+							"header": {
+								"MUlReo8L": [ "AUZG7wHG", "gsN0Dc2N" ],
+								"1UJXjVrT": [ "OJgxzTfk", "xZZrFsq7" ]
+							},
+							"method": "5wkAxCUE",
+							"tcp": "MN3oA9D2",
+							"interval": "32718s",
+							"docker_container_id": "cU15LMet",
+							"shell": "nEz9qz2l",
+							"tls_skip_verify": true,
+							"timeout": "34738s",
+							"ttl": "22773s",
+							"deregister_critical_service_after": "84282s"
+						},
+						{
+							"id": "UHsDeLxG",
+							"name": "PQSaPWlT",
+							"notes": "jKChDOdl",
+							"status": "5qFz6OZn",
+							"script": "PbdxFZ3K",
+							"http": "1LBDJhw4",
+							"header": {
+								"cXPmnv1M": [ "imDqfaBx", "NFxZ1bQe" ],
+								"vr7wY7CS": [ "EtCoNPPL", "9vAarJ5s" ]
+							},
+							"method": "wzByP903",
+							"tcp": "2exjZIGE",
+							"interval": "5656s",
+							"docker_container_id": "5tDBWpfA",
+							"shell": "rlTpLM8s",
+							"tls_skip_verify": true,
+							"timeout": "4868s",
+							"ttl": "11222s",
+							"deregister_critical_service_after": "68482s"
+						}
+					]
+				}
+			],
+			"session_ttl_min": "26627s",
+			"skip_leave_on_interrupt": true,
+			"start_join": [ "LR3hGDoG", "MwVpZ4Up" ],
+			"start_join_wan": [ "EbFSc3nA", "kwXTh623" ],
+			"syslog_facility": "hHv79Uia",
+			"tagged_addresses": {
+				"7MYgHrYH": "dALJAhLD",
+				"h6DdBy6K": "ebrr9zZ8"
+			},
+			"telemetry": {
+				"circonus_api_app": "p4QOTe9j",
+				"circonus_api_token": "E3j35V23",
+				"circonus_api_url": "mEMjHpGg",
+				"circonus_broker_id": "BHlxUhed",
+				"circonus_broker_select_tag": "13xy1gHm",
+				"circonus_check_display_name": "DRSlQR6n",
+				"circonus_check_force_metric_activation": "Ua5FGVYf",
+				"circonus_check_id": "kGorutad",
+				"circonus_check_instance_id": "rwoOL6R4",
+				"circonus_check_search_tag": "ovT4hT4f",
+				"circonus_check_tags": "prvO4uBl",
+				"circonus_submission_interval": "DolzaflP",
+				"circonus_submission_url": "gTcbS93G",
+				"disable_hostname": true,
+				"dogstatsd_addr": "0wSndumK",
+				"dogstatsd_tags": [ "3N81zSUB","Xtj8AnXZ" ],
+				"filter_default": true,
+				"prefix_fi
```

Comment:
Seems like we can eventually divorce the runtime config from the structure with both check and checks.